internal/documentdb/pool_cursors.go

```diff
@@ -222,3 +222,32 @@ func (p *Pool) ListIndexes(ctx context.Context, db string, spec wirebson.RawDocu
 
 	return page, cursorID, nil
 }
+
+// persistConn returns true if the connection should be persisted.
+// The zero cursor ID or empty continuation should not persist the connection.
+// It logs where persist is true but continuation is empty or cursorID is zero.
+func (p *Pool) persistConn(persist bool, cursorID int64, continuation wirebson.RawDocument) bool {
```

Comment:
TBH, I would expect method `persistConn` to persist conn, but it doesn't do that. And then there is a boolean flag `persist`…

A better method name is probably `shouldPersist` – you even use that word in the comment, and it is often a good idea to rename a method once you know what it is doing after writing the comment 😄 

And is the whole method basically a workaround? We should add an issue link, then.